dhcpv4/nclient4/client.go

```diff
@@ -447,12 +454,20 @@ func (c *Client) Request(ctx context.Context, modifiers ...dhcpv4.Modifier) (off
 		return
 	}
 
-	ack, err = c.SendAndRead(ctx, c.serverAddr, request, nil)
+	ack, err := c.SendAndRead(ctx, c.serverAddr, request, nil)
 	if err != nil {
 		err = fmt.Errorf("got an error while processing the request: %w", err)
 		return
 	}
-
+	lease = &Lease{}
+	lease.ACK = ack
+	lease.Offer = offer
+	lease.CreationTime = time.Now()
+	lease.IDOptions = dhcpv4.Options{}
+	for _, optioncode := range c.clientIDOptions {
+		v := request.Options.Get(optioncode)
+		lease.IDOptions.Update(dhcpv4.OptGeneric(optioncode, v))
```

Comment:
Are you copying this so that it can later be applied to the release? How about copying the modifiers instead?

Reply:
yes, that's purpose; however in order to make this more generic, for example if user wants to save lease to a file, it is better to copy the option itself, since modifier is just a function;